modules/custom/az_publication/src/AZPublicationTypeListBuilder.php

```diff
@@ -0,0 +1,220 @@
+    if ($this->limit) {
+      $query->pager($this->limit);
+    }
+    return $query->execute();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDefaultOperations(EntityInterface $entity) {
+    $operations = parent::getDefaultOperations($entity);
+    $current_user = \Drupal::currentUser();
+    if (!$current_user->hasPermission('delete publication type entities') && isset($operations['delete'])) {
+      unset($operations['delete']);
+    }
+    if (!$current_user->hasPermission('disable publication type entities') && isset($operations['disable'])) {
+      unset($operations['disable']);
+    }
+    if (!$current_user->hasPermission('enable publication type entities') && isset($operations['enable'])) {
+      unset($operations['enable']);
+    }
+    // Ensure ajax.js focuses on appropriate element by setting
+    // data-drupal-selector.
+    foreach ($operations as &$operation) {
+      $operation['attributes']['data-drupal-selector'] = 'az-publication-type-listing-' . $entity->id();
+    }
+
+    return $operations;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function load() {
+    $entities = [
+      'enabled' => [],
+      'disabled' => [],
+    ];
+    foreach (parent::load() as $entity) {
+      /** @var \Drupal\az_publication\Entity\AZPublicationTypeInterface $entity */
+      if ($entity->get('status')) {
+        $entities['enabled'][] = $entity;
+      }
+      else {
+        $entities['disabled'][] = $entity;
+      }
+    }
+    return $entities;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildRow(EntityInterface $entity) {
+    /** @var \Drupal\az_publication\Entity\AZPublicationTypeInterface $entity */
+    $row = parent::buildRow($entity);
+    return [
+      'data' => [
+        'id' => [
+          'data' => [
+            '#plain_text' => $entity->id(),
+          ],
+        ],
+        'label' => [
+          'data' => [
+            '#plain_text' => $entity->label(),
+          ],
+        ],
+        'type' => [
+          'data' => [
+            '#plain_text' => $entity->get('type'),
+          ],
+        ],
+        'operations' => $row['operations'],
+      ],
+      '#attributes' => [
+        'class' => [$entity->get('status') ? 'az-publication-ui-list-enabled' : 'az-publication-ui-list-disabled'],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildHeader() {
+    return [
+      'id' => [
+        'data' => $this->t('Machine name'),
+        'field' => 'id',
+        'specifier' => 'id',
+        '#attributes' => [
+          'class' => [
+            RESPONSIVE_PRIORITY_LOW,
+            'az-publication-type-machine-name',
+          ],
+        ],
+      ],
+      'label' => [
+        'data' => $this->t('Label'),
+        'field' => 'label',
+        'specifier' => 'label',
+        '#attributes' => [
+          'class' => [
+            RESPONSIVE_PRIORITY_LOW,
+            'az-publication-type-label',
```

Comment:
I think it might be best to have the Label column first, then the machine name:

```suggestion
        'data' => $this->t('Label'),
        'field' => 'label',
        'specifier' => 'label',
        '#attributes' => [
          'class' => [
            RESPONSIVE_PRIORITY_LOW,
            'az-publication-type-label',
          ],
        ],
      ],
      'label' => [
        'data' => $this->t('Machine name'),
        'field' => 'id',
        'specifier' => 'id',
        '#attributes' => [
          'class' => [
            RESPONSIVE_PRIORITY_LOW,
            'az-publication-type-machine-name',
```